scripts/redact_cli_py/redact/preprocess/pdf_renderer.py

```diff
@@ -3,79 +3,41 @@
 # root for license information.
 
 from PIL import Image
-
-import ctypes
-import pypdfium as pdfium
+import pypdfium2 as pdfium
 
 WHITE = 0xFFFFFFFF
 
 
 class PdfRenderer:
+    
+    # TODO load PDFium document only once on class level
```

Comment:
@mara004 question: what does this TODO mean? Is it already done by you somewhere else?

Reply:
No, it's not done yet because this descends a bit deeper into the functionality:
Currently, you're loading the document repeatedly (in `get_page_count()` and in `render_pdf()`), which isn't ideal. Instead, you should load it only once and close once the `PdfRenderer` object isn't needed anymore.
But that would likely result in an API-breaking change and I don't know if this part is considered public.

Also note that you're probably better off with pypdfium2's built-in multipage renderer `PdfDocument.render_to()` which works concurrently and is thus much faster. But that, too, would result in an API-breaking change.

Reply:
Please let me know if you don't mind breaking changes to the PdfRenderer API, then I can make an attempt to implement this (either here or in a new PR, as you like).

Reply:
**tl;dr** I don't mind breaking changes to most of the API in this folder (redact_cli_py/). This is in the very early stage of the pre-1.0 version. 
I guess keeping everything related in this PR is an easier way for me and @buddhawang. Thank you for your thoughtfulness!

Reply:
Good, that simplifies matters! Though it might take some time until I can work on this again.